### PR TITLE
Fix layer ordering between dropdown and breadcrumbs backgrounds. 

### LINKED ISSUE #6371: Can not click on a dropdown to change execution mode if the mouse pointer is above the text
### Discord username

_No response_

### What type of issue is this?

Permanent – Occurring repeatably

### Is this issue blocking you from using Enso?

- [ ] Yes, I can't use Enso because of this issue.

### Is this a regression?

- [ ] Yes, previous version of Enso did not have this issue.

### What issue are you facing?

When I am trying to change the execution mode and click on a text with the name the dropdown is not working. I have to click on the green space next to the name.

### Expected behaviour

The whole button should be clickable.

### How we can reproduce it?

_No response_

### Screenshots or screencasts

_No response_

### Enso Version

nightly 20.04.2023

### Browser or standalone distribution

Standalone distribution

### Browser Version or standalone distribution

standalone

### Operating System

MacOS

### Operating System Version

_No response_

### Hardware you are using

_No response_

### PR DESCRIPTION
**Pull request by MichaelMauderer**
### Pull Request Description

This PR fixes #6371.

### Important Notes

@kazcw @wdanilo I don't particularly like this solution, but I don't see any other good way to define the relationship between two instances of the same shape (`Rectangle`) used in different UI elements. If you are aware of a more elegant solution, I’d be happy to hear any suggestions. 

### QA Note

This should be tested on Windows, Linux, and Mac. 

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] ~Unit tests have been written where possible.~
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: lib/rust/ensogl/core/src/display/scene.rs [hunk 1]
```diff
@@ -597,29 +610,31 @@ impl Renderer {
...
+    pub above_nodes_text: Layer,
     /// `panel` layer contains all panels with fixed position (not moving with the panned scene)
     /// like status bar, breadcrumbs or similar.
-    pub panel_background:   Layer,
-    pub panel:              Layer,
-    pub panel_text:         Layer,
-    pub node_searcher:      Layer,
+    pub panel_background_rect_level_0: RectLayerPartition,
+    pub panel_background_rect_level_1: RectLayerPartition,
+    pub panel_background: Layer,
```
Comment (farmaazon) [inline lib/rust/ensogl/core/src/display/scene.rs]: The docs were for `panel_background` field before. Also add docs for rect_level_0 as it's very unclear what is the purpose of those.

File: lib/rust/ensogl/core/src/display/scene.rs [hunk 2]
```diff
@@ -589,6 +591,17 @@ impl Renderer {
...
+
+
+fn partition_layer<S: display::shape::primitive::system::Shape>(
+    base_layer: &Layer,
+    name: &str,
+) -> Rc<LayerSymbolPartition<S>> {
+    let partition = base_layer.create_symbol_partition::<S>(name);
+    Rc::new(partition)
+}
+
```
Comment (farmaazon) [inline lib/rust/ensogl/core/src/display/scene.rs:605]: I wonder if this function is worth keeping. We could "inline" it, simplifying the code readability (one function name less to remember).

**Review by farmaazon — APPROVED** [review]

**Comment by kazcw** [comment]
> I don't particularly like this solution, but I don't see any other good way to define the relationship between two instances of the same shape (`Rectangle`) used in different UI elements. If you are aware of a more elegant solution, I’d be happy to hear any suggestions. 

Yes, the current interface to layer symbol partitions (i.e. exposing them directly) is very low-level. During the next draw-calls push I intend to put it behind a more convenient abstraction (that could eventually be used to unify the 3 layering mechanisms, and instead distinguish inside-component relationships and between-component relationships). Until then, hard-coded partitions are ugly but they save draw calls.

File: lib/rust/ensogl/core/src/display/scene.rs (hunk 1 above)
Comment (kazcw) [reply lib/rust/ensogl/core/src/display/scene.rs]: I think the original comment had become unclear with previous changes---It is meant to refer to the group of `panel_` layers.

File: lib/rust/ensogl/core/src/display/scene.rs (hunk 2 above)
Comment (MichaelMauderer) [reply lib/rust/ensogl/core/src/display/scene.rs:605]: It greatly declutters the usage, though. Especially as this is easily results in “too long lines”. I would opt for keeping it.

File: lib/rust/ensogl/core/src/display/scene.rs (hunk 1 above)
Comment (MichaelMauderer) [reply lib/rust/ensogl/core/src/display/scene.rs]: I will update the comment to more clearly refer to the group of layers and partitions.

**Comment by farmaazon** [comment]
## QA on Linux: :green_circle: 

No issues found.

**Comment by vitvakatu** [comment]
QA on Mac OS: 🟢

**Pull request merged**